xla/service/cpu/onednn_matmul_rewriter.cc

```diff
@@ -375,9 +382,13 @@ bool OneDnnMatMulRewriter::ShouldRewrite(const HloInstruction* dot_instr) {
 
   // Layout should be row-major, contraction dimensions captures transpose
   // scenarios in last two dimensions.
-  if (!IsRowMajor(lhs_shape) || !IsRowMajor(rhs_shape) ||
-      !IsRowMajor(output_shape)) {
-    return false;
+  // Col-major layouts are corrected to row-majow for BatchDot operation as
+  // part of the layout-pass.
+  if (!IsBatchDot(*dot_instr)) {
+    if (!IsRowMajor(lhs_shape) || !IsRowMajor(rhs_shape) ||
+        !IsRowMajor(output_shape)) {
+      return false;
+    }
   }
```

Comment:
Nit: Let's avoid nested `if`s. (Please apply code formatter.)
```suggestion
  if (!IsBatchDot(*dot_instr) && 
     (!IsRowMajor(lhs_shape) || !IsRowMajor(rhs_shape) || 
          !IsRowMajor(output_shape))) {
    return false;
  }
```

Reply:
@penpornk I made the requested change. Thanks!